src/main/java/uk/gov/companieshouse/chdorderconsumer/service/ItemOrderedProcessorService.java

```diff
@@ -51,8 +51,15 @@ MissingImageDeliveryRequestApi mapChdItemOrderedToMissingImageDeliveryRequestApi
         missingImageDeliveryRequestApi.setFilingHistoryBarcode(itemOptions.get("filingHistoryBarcode"));
         missingImageDeliveryRequestApi.setItemCost(item.getTotalItemCost());
 
-        String entityID = mongoService.getEntityId(itemOptions.get("filingHistoryId"));
-        missingImageDeliveryRequestApi.setEntityId(entityID);
+        String filingHistoryId = itemOptions.get("filingHistoryId");
+
+        missingImageDeliveryRequestApi.setEntityId(mongoService.getEntityId(filingHistoryId));
+
+        if (missingImageDeliveryRequestApi.getEntityId() == null &&
+            missingImageDeliveryRequestApi.getFilingHistoryBarcode() == null) {
+
+            missingImageDeliveryRequestApi.setFilingHistoryBarcode(mongoService.getBarcode(filingHistoryId));
+        }
 
```

Comment:
Rather than calling the database twice, couldn't we call it once with a Projections.include("_entity_id","barcode") type of restriction and then assign the entity id and the chs barcode etc to a java object and reference them from there if needed?

e.g. 
`FilingHistory filingHistory = mongoService.queryMongo(filingHistoryId);
if(filingHistory.getEntityId() == null && missingImageDeliveryRequestApi.getFilingHistoryBarcode() == null){
    missingImageDeliveryRequestApi.setFilingHistoryBarcode(filingHistory.getBarcode);
}`

Also, do we need to check if the call to mongoService.getBarcode on line 61 is null?

Reply:
I thought it best to keep it separate as it is only ever called when there is no `_entity_id` and no `_barcode`. I think it is a little easier to read also keeping the setting of the values in the `ItemOrderedProcessorService` and explicitly saying what the method does i.e `getEntityId` and `getBarcode`.

In regard to to if we need a null check in line 61, its ok as a null is an acceptable value similar to when we check for the entity id.